Release Notes/510.md

```diff
@@ -48,12 +48,22 @@
   - Issue: https://github.com/apple/swift-syntax/issues/2092
   - Pull Request: https://github.com/apple/swift-syntax/pull/2184
 
+- `IncrementalParseTransition`:
+  - Description: The initializer `IncrementalParseTransition.init(previousTree:edits:lookaheadRanges:reusedNodeCallback:)` is marked as deprecated. The deprecated initializer emits a warning indicating the use of `IncrementalParseTransition.init(previousIncrementalParseResult:edits:reusedNodeCallback:)`.
+  - Issue: https://github.com/apple/swift-syntax/issues/2267
+  - Pull request: https://github.com/apple/swift-syntax/pull/2272
+
 ## API-Incompatible Changes
 
 - Effect specifiers:
   - Description: The `unexpectedAfterThrowsSpecifier` node of the various effect specifiers has been removed.
   - Pull request: https://github.com/apple/swift-syntax/pull/2219
 
+- `Parser.parseIncrementally(source:parseTransition:)` and `Parser.parseIncrementally(source:maximumNestingLevel:parseTransition:)`:
+  - Description: `Parser.parseIncrementally(source:parseTransition:)` and `Parser.parseIncrementally(source:maximumNestingLevel:parseTransition:)` now have new return type named `IncrementalParseResult`.
```

Comment:
```suggestion
  - Description: The default versions of `Parser.parseIncrementally` return a `IncrementalParseResult` instead of a tuple. Access to the struct should be compatible with the tuple in almost all cases unless the tuple is stored into a variable and then destructed or passed to a function that expects a tuple.
```